modules/siddhi-core/src/main/java/org/wso2/siddhi/core/aggregation/IncrementalDataPurging.java

```diff
@@ -0,0 +1,366 @@
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class implements the logic which is needed to purge data which are related to incremental
+ **/
+public class IncrementalDataPurging implements Runnable {
+    private static final Logger LOG = Logger.getLogger(IncrementalDataPurging.class);
+    private long purgeExecutionInterval = Expression.Time.minute(5).value();
+    private long sec = Expression.Time.sec(30).value();
+    private long mins = Expression.Time.hour(24).value();
+    private long hours = Expression.Time.day(30).value();
+    private long days = Expression.Time.year(5).value();
+    private long months = -1;
+    private long years = -1;
+    private boolean purgingEnabled = true;
+    private Map<TimePeriod.Duration, Long> retentionPeriods = new EnumMap<>(TimePeriod.Duration.class);
+    private StreamEventPool streamEventPool;
+    private Map<TimePeriod.Duration, Table> aggregationTables;
+    private SiddhiAppContext siddhiAppContext;
+    private Map<String, Table> tableMap;
+    private static ScheduledFuture scheduledFuture;
+    private static final String INTERNAL_AGG_TIMESTAMP_FIELD = "AGG_TIMESTAMP";
+
+
+    public void init(AggregationDefinition aggregationDefinition, StreamEventPool streamEventPool,
+                     Map<TimePeriod.Duration, Table> aggregationTables, SiddhiAppContext siddhiAppContext,
+                     Map<String, Table> tableMap) {
+        this.siddhiAppContext = siddhiAppContext;
+        List<Annotation> annotations = aggregationDefinition.getAnnotations();
+        this.tableMap = tableMap;
+        this.streamEventPool = streamEventPool;
+        this.aggregationTables = aggregationTables;
+        for (TimePeriod.Duration duration : aggregationTables.keySet()) {
+            switch (duration) {
+                case SECONDS:
+                    retentionPeriods.put(duration, sec);
+                    break;
+                case MINUTES:
+                    retentionPeriods.put(duration, mins);
+                    break;
+                case HOURS:
+                    retentionPeriods.put(duration, hours);
+                    break;
+                case DAYS:
+                    retentionPeriods.put(duration, days);
+                    break;
+                case MONTHS:
+                    retentionPeriods.put(duration, months);
+                    break;
+                case YEARS:
+                    retentionPeriods.put(duration, years);
+            }
+        }
+
+        Map<String, Annotation> annotationTypes = new HashMap<>();
+
+        for (Annotation annotation : annotations) {
+            annotationTypes.put(annotation.getName().toLowerCase(), annotation);
+        }
+        if (annotationTypes.keySet().contains(SiddhiConstants.NAMESPACE_PURGE)) {
+            if (Objects.nonNull(annotationTypes.get(SiddhiConstants.NAMESPACE_PURGE).
```

Comment:
We can move this nonNull check to the if in line 119 as the first condition

Reply:
If we do so, "else" case will get hit when the annotation element "enable" is not defined case as well, since this is an optional element it is acceptable. isn't it?